roachpb/data.go

```diff
@@ -451,22 +493,31 @@ func (v Value) GetTimeseries() (InternalTimeSeriesData, error) {
 	return ts, v.GetProto(&ts)
 }
 
+var crc32Pool = sync.Pool{
+	New: func() interface{} {
+		return crc32.NewIEEE()
+	},
+}
+
 // computeChecksum computes a checksum based on the provided key and
-// the contents of the value. If the value contains a byte slice, the
-// checksum includes it directly.
+// the contents of the value.
 func (v Value) computeChecksum(key []byte) uint32 {
+	if len(v.RawBytes) < headerSize {
+		return 0
+	}
 	crc := crc32Pool.Get().(hash.Hash32)
 	if _, err := crc.Write(key); err != nil {
 		panic(err)
 	}
-	if v.RawBytes != nil {
-		if _, err := crc.Write(v.RawBytes); err != nil {
-			panic(err)
-		}
+	if _, err := crc.Write(v.RawBytes[4:]); err != nil {
+		panic(err)
 	}
 	sum := crc.Sum32()
 	crc.Reset()
 	crc32Pool.Put(crc)
+	if sum == 0 {
```

Comment:
I should comment this. Previously we used the presence of the `Value.Checksum` field to determine if the checksum had been set. Now we reserve the checksum value of 0 to indicate the checksum has not been initialized. We ensure that the computation of the checksum never produces a 0 value by folding 0 into 1.